ledger/src/mask/mask_impl.rs

```diff
@@ -27,6 +27,7 @@ pub enum MaskImpl {
     Attached {
         parent: Mask,
         owning_account: HashMap<AccountIndex, Account>,
+        token_owners: HashMap<TokenId, AccountId>,
```

Comment:
Can you make it an `Option<HashMap<TokenId, AccountId>>`, and optionally call `derive_token_id` in the diff of this file